### PR TITLE
Feature/secondary ip

### PR DESCRIPTION
**Pull request by grenader**
1. **What this PR does / why we need it**:
Add support for the 'Secondary IP' feature.

We have added:
- two new fields: PodSubnetID and NetworkPluginMode 
- various validations to make sure the correct information was passed to MKP
- unit tests to verify extended valuation 

The validations are:
- Network-plugin-mode 'overlay' can be only set when network-plugin is 'azure' - DONE 
- If network-plugin-mode 'overlay' is selected pod-subnet-id should NOT be defined in node pools.
- Node (Vnet-subnet) and pod subNets should belong to the same vNet
- pod subNet cannot be specified if node subNet is not defined
- node (Vnet-subnet) and pod subNets should be different from each other, this is in every node pool and across the whole cluster
- podCIDR can be only set for network-plugin is 'kubenet' or network-plugin is 'azure' with network-plugin-mode is 'overlay

2. **Which issue(s) this PR fixes**
(https://jira.eng.vmware.com/browse/TMC-43293)

3. **Additional information**

In addition to the 'secondary IP' field validation, the 
"DNS Server Ip or service CIDR cannot be set is network plugin is 'kubenet'" restriction has been removed, to follow Azure's actual behavior.

### CODE REVIEW

File: internal/models/akscluster/nodepool_spec.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ type VmwareTanzuManageV1alpha1AksclusterNodepoolSpec struct {
 	// If this is not specified, a VNET and subnet will be generated and used. If no podSubnetID is specified, this applies to
 	// nodes and pods, otherwise it applies to just nodes. This is of the form:
 	VnetSubnetID string `json:"vnetSubnetId,omitempty"`
+
+	//Todo: change this
```
Comment (ramya-bangera) [inline internal/models/akscluster/nodepool_spec.go:83]: TODO?

File: README.md [hunk 2]
```diff
@@ -211,6 +211,14 @@ The plugin will read the descriptions and schema of each resource and data sourc
 
 Please refer to `examples` folder to perform CRUD operations with Tanzu Mission Control provider for various resources
 
+# Troubleshooting
+
+## Executions of a different version of the provider
+Terraform will always look for the latest version of the provided and will use it even if you have just built a previous version.
```
Comment (ramya-bangera) [inline README.md:217]: nit: version of the provided -> version of the provider

File: internal/models/akscluster/nodepool_spec.go (hunk 1 above)
Comment (grenader) [reply internal/models/akscluster/nodepool_spec.go:83]: Good catch, I wanted to update these comments

File: README.md (hunk 2 above)
Comment (grenader) [reply README.md:217]: fixed

**Comment by grenader** [comment]
> Could you update the [examples ](https://github.com/vmware/terraform-provider-tanzu-mission-control/tree/main/examples/data-sources), [templates](https://github.com/vmware/terraform-provider-tanzu-mission-control/tree/main/resource_templates) and acceptance tests with these changes?

Thank you, @gargi. I have updated examples, templates, and regenerated documentation.
I do not think we have acceptance tests here. At least, I do not see tests that would make external calls.

**Review by gshaw-pivotal — DISMISSED** [review]

**Comment by asutoshpalai** [comment]
Wow, this was painful to review because of so many old commits in the MR! Please them up next time or rebase or merge the main into it.

I tried to review as many files as I thought were relevant to the topic, and they look good.

My approval is a partial rubber stamp as I am not sure I reviewed all the required files.

**Review by asutoshpalai — DISMISSED** [review]

**Comment by grenader** [comment]
I hear you @asutoshpalai. This feature work happened at the same time as many other changes, and I had to I had to rebase several times. That extended the list of changes.

**Comment by grenader** [comment]
I am wondering why "grenader dismissed asutoshpalai’s stale review 2 hours ago" happened. I have not committed any new changes.

**Comment by grenader** [comment]
I have created a cleaner copy of these changes. 
https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/304

**Pull request closed without merging**